### ISSUE TITLE
Using moduleResolution NodeNext breaks type definitions for importing

### ISSUE THREAD

**Issue by rachaeldawn** [effector/effector#757]
Typescript reports that there are no type definitions when using `NodeNext` for `"moduleResolution": "NodeNext"`, but not `"moduleResolution": "node"`. This is true on the `22.3.0` build.

Please see [this repo](https://github.com/rachaeldawn/effector-types-bug) for a minimal reproduction.

I am currently working on a fix, just verifying. [This is the repo](https://github.com/rachaeldawn/effector) I am using for the fix. Check the branch `fix-demo` on the reproduction to see fixes in action.

I don't know what the process is for creating a PR. Could someone direct me to how to go about that?

**Comment by rachaeldawn** [comment]
Forgot to update this. The fix was completed and should be ready for a PR.

**Comment by igorkamyshev** [comment]
As I understand, it was fixed by #759? May I close it?

**Comment by rachaeldawn** [comment]
> As I understand, it was fixed by #759? May I close it?

Yes ☺️ I believe it is still pending release, but yes, it is resolved
